### PR TITLE
task-2

### PR DESCRIPTION
**Pull request by bohdan-lysenko-lohika**

### CODE REVIEW

File: task-2/app.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        }
+        const { handler } = router;
+        handler(request, response, parsedUrl, bodyAsJson);
+    } catch(e) {
+        console.log(e);
+        sendBadRequestResponse(response, 'please provide valid arguments');
+    }
+});
+
+server.on(SERVER_EVENT_ERROR, (err) => console.log(`Error: ${err.message}\n${err.stack}`));
```
Comment (viktornord) [inline task-2/app.js:31]: here `console.error` should be used instead of `console.log`

File: task-2/response-helper.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
+const RESPONSE_HEADER_CONTENT_TYPE = 'content-type';
```
Comment (viktornord) [inline task-2/response-helper.js:1]: good, but next time consider this https://nodejs.org/api/http.html#http_http_status_codes

File: task-2/response-helper.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    response.statusCode = STATUS_OK;
+    response.statusMessage = STATUS_OK_MESSAGE;
+    response.end(JSON.stringify(data));
+}
+
+function sendNotFoundResponse(response) {
+    setBasicHeaders(response);
+    response.statusCode = STATUS_NOT_FOUND;
+    response.statusMessage = STATUS_NOT_FOUND_MESSAGE;
+    response.end(JSON.stringify(null));
```
Comment (viktornord) [inline task-2/response-helper.js:24]: no need to stringify `null`

File: task-2/request-helper.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        request.on(REQUEST_EVENT_DATA_END, () => {
+            try {
+                if (!bodyChunks.length) {
+                    resolve(null);
+                    return;
+                }
+                const body = JSON.parse(Buffer.concat(bodyChunks));
+                resolve(body);
+            } catch(e) {
+                console.log(e);
```
Comment (viktornord) [inline task-2/request-helper.js:17]: console.error

File: task-2/routes/sum.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
+const { sendSuccessResponse, sendBadRequestResponse } = require('./../response-helper');
+
+function handler(request, response, url, data) {
+    const { a } = url.query;
+    const { b } = data;
+    const nA = Number(a);
+    const nB = Number(b);
+    if (isNaN(nA) || isNaN(nB)) {
```
Comment (viktornord) [inline task-2/routes/sum.js:8]: you can use `Number.isNaN`

Comment (lysenkobv) [reply task-2/routes/sum.js]: ahh, totally forgot about that, thank you

Comment (bohdan-lysenko-lohika) [reply task-2/routes/sum.js:8]: ahh, totally forgot about that, thank you

**Pull request merged**